app/models/campaign.rb

```diff
@@ -100,7 +100,7 @@ def validate_dates
   # Intercept Rails typecasting and add error if given value cannot be parsed into a date.
   def validate_date_attribute(date_type)
     value = send("#{date_type}_before_type_cast")
-    self[date_type] = value.is_a?(Date) || value.is_a?(Time) ? value : DateTime.parse(value)
+    self[date_type] = value.is_a?(Date) || value.is_a?(Time) ? value : Date.parse(value)
```

Comment:
We should probably use Time here, as it's more directly a substitute for DateTime.

Reply:
Hi, the parse function we are using here is not available in Time class in ruby-2.5.0, as far as I know, I tried searching for it but couldn't find an equivalent function and thus went with Date instead.

Reply:
It's there: https://ruby-doc.org/stdlib-2.5.0/libdoc/time/rdoc/Time.html#method-c-parse

If you're working in `irb`, then you won't get that functionality unless you `require 'time'`. But in Rails, it looks like it's already loaded and will just work.